google-cloud-datastore/src/main/java/com/google/cloud/datastore/Transaction.java

```diff
@@ -176,6 +176,205 @@
   <T> QueryResults<T> run(Query<T> query);
 
   /**
+   * Datastore add operation. This method will also allocate id for any entity with an incomplete
+   * key. As opposed to {@link #add(FullEntity)} and {@link #add(FullEntity...)}, this method will
+   * defer any necessary id allocation to commit time.
```

Comment:
Seems unclear.   Perhaps:
**This method will also allocate id for any entity with an incomplete key unlike {@link #add(FullEntity)} and {@link #add(FullEntity...)}, this method defers any necessary id allocation to commit time.**